storagemarket/deal_execution.go

```diff
@@ -148,6 +157,13 @@ func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter,
 			return &dealMakingError{err: fmt.Errorf("error when matching commP for imported data for offline deal: %w", err)}
 		}
 		p.dealLogger.Infow(deal.DealUuid, "commp matched successfully for imported data for offline deal")
+
+		// update checkpoint
+		if err := p.updateCheckpoint(pub, deal, dealcheckpoints.Transferred); err != nil {
+			return &dealMakingError{
+				err: fmt.Errorf("failed to update checkpoint: %w", err),
```

Comment:
what happens if we don't do this ?

Reply:
Then we would run commp each time boost restarts